sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -209,6 +216,95 @@ namespace Azure { namespace Core { namespace Http {
         NextHttpPolicy nextHttpPolicy) const override;
   };
 
+  class ValuePolicy;
+
+  /**
+   * @brief @ValuePolicy options.
+   */
+  class ValuePolicyOptions {
+    friend class ValuePolicy;
+
+  private:
+    std::map<std::string, std::string> m_headerValues;
+    std::map<std::string, std::string> m_queryValues;
+
+  public:
+    /**
+     * @brief Add values (key-value pairs) that shall be applied to HTTP request headers.
+     * @detail If a value with the a key from \p headerValues already exists in this @ValuePolicy,
+     * the new value from \p headerValues will be taken.
+     * @note Header keys are case-insensitive.
+     * @param headerValues Values that shall be applied.
+     */
+    void AddHeaderValues(std::map<std::string, std::string> const& headerValues)
```

Comment:
We want options structs to be immutable after creation so that they are thread safe.
So, I'd have the ctor take the two maps as arguments and NOT have these 2 AddXxx functions.